saleor/core/search_tasks.py

```diff
@@ -55,23 +50,54 @@ def set_order_search_document_values(total_count, updated_count):
         task_logger.info("No orders to update.")
         return
 
-    orders = []
-    for order in qs:
-        order.search_document = prepare_order_search_document_value(
-            order, already_prefetched=True
-        )
-        orders.append(order)
-    Order.objects.bulk_update(orders, ["search_document"])
-
-    updated_count += len(orders)
-    progress = round((updated_count / total_count) * 100, 2)
-
-    task_logger.info(
-        f"Updated {updated_count} from {total_count} orders - {progress}% done."
+    updated_count = set_search_document_values(
+        qs, total_count, updated_count, prepare_order_search_document_value
     )
 
     if updated_count == total_count:
         task_logger.info("Setting order search document values finished.")
         return
 
     return set_order_search_document_values.delay(total_count, updated_count)
+
+
+@app.task
+def set_product_search_document_values(total_count, updated_count):
+    qs = Product.objects.filter(search_document="").prefetch_related(
+        *PRODUCT_FIELDS_TO_PREFETCH
+    )[:BATCH_SIZE]
```

Comment:
⚠️ shouldn't this be iterated over in batches instead of limiting it to batch size?